src/resspect/database.py

```diff
@@ -249,42 +255,15 @@ def load_features_from_file(self, path_to_features_file: str, screen=False,
 
-        #! Make this part work better with the different feature extractors
-        # list of features to use
-        if survey == 'DES':
-            if feature_extractor == "Bazin":
-                self.features_names = ['gA', 'gB', 'gt0', 'gtfall', 'gtrise', 'rA',
-                                       'rB', 'rt0', 'rtfall', 'rtrise', 'iA', 'iB',
-                                       'it0', 'itfall', 'itrise', 'zA', 'zB', 'zt0',
-                                       'ztfall', 'ztrise']
-            elif feature_extractor == 'Bump':
-                self.features_names = ['gp1', 'gp2', 'gp3', 'gmax_flux', 
-                                       'rp1', 'rp2', 'rp3', 'rmax_flux', 
-                                       'ip1', 'ip2', 'ip3', 'imax_flux', 
-                                       'zp1', 'zp2', 'zp3', 'zmax_flux']
-            elif feature_extractor == 'Malanchev':
-                self.features_names = ['ganderson_darling_normal','ginter_percentile_range_5',
-                                       'gchi2','gstetson_K','gweighted_mean','gduration', 
-                                       'gotsu_mean_diff','gotsu_std_lower', 'gotsu_std_upper',
-                                       'gotsu_lower_to_all_ratio', 'glinear_fit_slope', 
-                                       'glinear_fit_slope_sigma','glinear_fit_reduced_chi2',
-                                       'randerson_darling_normal', 'rinter_percentile_range_5',
-                                       'rchi2', 'rstetson_K', 'rweighted_mean','rduration', 
-                                       'rotsu_mean_diff','rotsu_std_lower', 'rotsu_std_upper',
-                                       'rotsu_lower_to_all_ratio', 'rlinear_fit_slope', 
-                                       'rlinear_fit_slope_sigma','rlinear_fit_reduced_chi2',
-                                       'ianderson_darling_normal','iinter_percentile_range_5',
-                                       'ichi2', 'istetson_K', 'iweighted_mean','iduration', 
-                                       'iotsu_mean_diff','iotsu_std_lower', 'iotsu_std_upper',
-                                       'iotsu_lower_to_all_ratio', 'ilinear_fit_slope', 
-                                       'ilinear_fit_slope_sigma','ilinear_fit_reduced_chi2',
-                                       'zanderson_darling_normal','zinter_percentile_range_5',
-                                       'zchi2', 'zstetson_K', 'zweighted_mean','zduration', 
-                                       'zotsu_mean_diff','zotsu_std_lower', 'zotsu_std_upper',
-                                       'zotsu_lower_to_all_ratio', 'zlinear_fit_slope', 
-                                       'zlinear_fit_slope_sigma','zlinear_fit_reduced_chi2']
+        # Create the filter-feature names based on the survey.
+        survey_filters = FILTER_SETS[survey]
+        self.features_names = create_filter_feature_names(survey_filters, feature_names)
```

Comment:
This is almost certainly insufficient. I believe that we need another method in the `LightCurve` base class that will allow us to return _just_ the feature names, excluding what is referred to as `metadata_names` here. 

Or more likely, we can break the methods into two, something like: 

```
def get_feature_header(...):
    return get_metadata_header() + get_features_header()

def get_metadata_header(...):
    # returns something like `['id', 'redshift', ...]

def get_features_header(...):
    # returns the cross product of features and filters
```

Reply:
Done